version: "3.8"

services:
  core:
    container_name: tracker_core
    build:
      context: .
      dockerfile: docker/tracker_core/core.Dockerfile
    volumes:
      - ./tracker_core:/app
    networks:
      - core_db_network
    ports:
      - "8000:8000"
    depends_on:
      - db_core
    env_file:
      - .core.env
    restart: on-failure
    healthcheck:
      test: ["CMD", "curl", "--fail", "http://0.0.0.0:8000/admin/login/"]
      interval: 1m
      timeout: 5s
      retries: 2

  db_core:
    container_name: db_core
    image: postgres:14-alpine
    volumes:
      - db_core_volume:/var/lib/postgresql/data
    networks:
      - core_db_network
    ports:
      - "5432:5432"
    env_file:
      - .core.env

  pgadmin:
    container_name: pgadmin
    image: dpage/pgadmin4
    environment:
      PGADMIN_DEFAULT_EMAIL: pgadmin@example.com
      PGADMIN_DEFAULT_PASSWORD: pgadmin
    networks:
      - core_db_network
    ports:
      - 8080:80
    depends_on:
      - db_core


networks:
  core_db_network:

volumes:
  db_core_volume: